src/coreFeatures/Account.java

```diff
@@ -135,19 +141,37 @@ public BigDecimal getInterestRate() {
     public void setInterestRate(BigDecimal interestRate) {
         this.interestRate = interestRate;
     }
-
+/**
+     * Constructs an account with a name and initial balance.
+     * 
+     * @param name The name of the account.
+     * @param balance The initial balance.
+     */
     public Account(String name, double balance) {
         this.name = name;
         this.balance = BigDecimal.valueOf(balance);
     }
-
+    
+/**
+     * Constructs an account with a name, initial balance, and interest rate details.
```

Comment:
This can be simplified, no need to include hasInterestRate since it can be implied as true.